src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -37,18 +41,22 @@ public OneOrMany(ImmutableArray<T> many)
             _many = many;
         }
 
+        [MemberNotNullWhen(true, nameof(_one))]
+        private bool HasOne
```

Comment:
Not sure what you mean.

Reply:
Sorry, it looks like this is a case where we relaxed the analysis in a later version. We do have a behavior where if you return a bool constant we verify that variables referenced in NotNullWhen/MemberNotNullWhen have not-null flow state and otherwise give a warning on the return. It looks like nothing is wrong here, although it may be prudent to add a debug assert here.